Revert "Pin Node to 18.13.0" when build agents are upgraded to 18.14.2

When using Node 18.14.0 or 18.14.1 with npm 9.3.1, pipeline "autorest.csharp" fails with the following error:

```
error TS2688: Cannot find type definition file for 'mocha'.
  The file is in the program because:
    Entry point of type library 'mocha' specified in compilerOptions
```

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2171100&view=logs&j=45f9faef-ad3a-5235-6d52-a4da1b0ab954&s=45a4dd8c-d89a-566b-d89f-6724086326ec&t=9a8cf63c-b194-5fc6-b9a6-f445bb38c677&l=37

I verified this is due to https://github.com/nodejs/node/issues/46542, and is fixed in npm 9.4.2 (#3133).

Node 18.14.2 was released today and should fix this issue.  Since this repo already pinned to Node 18.13.0 (#3097), I think the best approach is to wait until our build agents have upgraded to Node 18.14.2, then revert #3097.

--- a/eng/pipelines/build.yml
+++ b/eng/pipelines/build.yml
@@ -50,9 +50,9 @@ stages:
               useGlobalJson: true
               performMultiLevelLookup: true
           - task: NodeTool@0
-            displayName: "Install Node 18.13.0"
+            displayName: "Install Node 18.x"
             inputs:
-              versionSpec: '18.13.0'
+              versionSpec: '18.x'
           - script: |
               npm ci
             displayName: "Install packages"

--- a/eng/pipelines/cadl-ci.yml
+++ b/eng/pipelines/cadl-ci.yml
@@ -22,7 +22,7 @@ variables:
   - name: skipComponentGovernanceDetection
     value: true
   - name: NodeVersion
-    value: '18.13.0'
+    value: '18.x'
   - name: VAR_ARTIFACT_NAME
     value: 'CadlCsharpEmitter'
   - name: VAR_BUILD_ARTIFACT_STAGING_DIRECTORY

--- a/eng/pipelines/check-code-generation.yml
+++ b/eng/pipelines/check-code-generation.yml
@@ -12,9 +12,9 @@ jobs:
     vmImage: windows-2022
   steps:
     - task: NodeTool@0
-      displayName: "Install Node 18.13.0"
+      displayName: "Install Node 18.x"
       inputs:
-        versionSpec: '18.13.0'
+        versionSpec: '18.x'
     - task: UseDotNet@2
       displayName: 'Use .NET Core SDK'
       inputs:

--- a/eng/pipelines/mgmt-mocktest.yml
+++ b/eng/pipelines/mgmt-mocktest.yml
@@ -44,9 +44,9 @@ stages:
               useGlobalJson: true
               performMultiLevelLookup: true
           - task: NodeTool@0
-            displayName: "Install Node 18.13.0"
+            displayName: "Install Node 18.x"
             inputs:
-              versionSpec: '18.13.0'
+              versionSpec: '18.x'
           - script: |
               npm ci
             displayName: "Install packages"

